src/ipfs_retrieval.py

```
@@ -86,11 +92,34 @@
+        if os.path.exists(cache_file):
+            try:
+                with open(cache_file, "r") as f:
+                    json_cid = json.load(
+                        f
+                    )  # <-- can raise JSONDecodeError if file is empty/corrupt
+                for entry in json_cid:
+                    if entry == ipns_key_str:
+                        return json_cid[entry]
+            except (KeyError, json.JSONDecodeError):
+                # We tried local, but it’s also invalid (bad JSON or missing key)
+                raise DatasetNotFoundError("Invalid dataset name")
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)